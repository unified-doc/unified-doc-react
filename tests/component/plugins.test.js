import React from 'react';
import renderer from 'react-test-renderer';
import highlight from 'rehype-highlight';

import DocComponent from '../../lib/component';
import { jsContent } from '../fixtures';

const sanitizeSchema = {
  attributes: {
    '*': ['className', 'style'],
    mark: ['dataMarkId', 'id'],
  },
  clobberPrefix: '',
};

const marks = [{ id: 'a', start: 9, end: 14, classNames: ['a', 'b'] }];

describe('plugins', () => {
  it('applies prePlugins', () => {
    const prePlugins = [[highlight]];
    const options = {
      content: jsContent,
      filename: 'doc.js',
      marks,
      prePlugins,
      sanitizeSchema,
    };
    const tree = renderer.create(<DocComponent options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies postPlugins', () => {
    const postPlugins = [[highlight]];
    const options = {
      content: jsContent,
      filename: 'doc.js',
      marks,
      postPlugins,
      sanitizeSchema,
    };
    const tree = renderer.create(<DocComponent options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
