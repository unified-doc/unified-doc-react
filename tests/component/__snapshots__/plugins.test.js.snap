// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins applies postPlugins 1`] = `
<div>
  <div>
    <pre
      className="language-js"
    >
      <code
        className="hljs language-js"
      >
        <span
          className="hljs-function"
        >
          <span
            className="hljs-keyword"
          >
            function
          </span>
           
          <span
            className="hljs-title"
          >
            hello
          </span>
          (
          <span
            className="hljs-params"
          />
          ) 
        </span>
        {
  
        <span
          className="hljs-keyword"
        >
          return
        </span>
         
        <span
          className="hljs-string"
        >
          "world"
        </span>
        ;
}
      </code>
    </pre>
  </div>
</div>
`;

exports[`plugins applies prePlugins 1`] = `
<div>
  <div>
    <pre
      className="language-js"
    >
      <code
        className="hljs language-js"
      >
        <span
          className="hljs-function"
        >
          <span
            className="hljs-keyword"
          >
            function
          </span>
           
          <span
            className="hljs-title"
          >
            <mark
              className="a b"
              data-mark-id="a"
              id="a"
            >
              hello
            </mark>
          </span>
          (
          <span
            className="hljs-params"
          />
          ) 
        </span>
        {
  
        <span
          className="hljs-keyword"
        >
          return
        </span>
         
        <span
          className="hljs-string"
        >
          "world"
        </span>
        ;
}
      </code>
    </pre>
  </div>
</div>
`;
